bluetooth/sql/functions/reader_status_history.sql

```diff
@@ -0,0 +1,122 @@
+-- FUNCTION: bluetooth.reader_status_history(date)
+
+-- DROP FUNCTION bluetooth.reader_status_history(date);
+
+CREATE OR REPLACE FUNCTION bluetooth.reader_status_history(
+	insert_value date)
+    RETURNS TABLE(an_id bigint, last_received date, status text, ddown integer, start_route_point_id text, start_detector text, end_route_point_id text, end_detector text)
+    LANGUAGE 'plpgsql'
+
+    COST 100
+    VOLATILE 
+AS $BODY$	
+	begin
+		--return query
+			with x AS 
+			(select DISTINCT(analysis_id), 
+			 CASE
+			 WHEN MAX (datetime_bin::date)>= (insert_value) then (insert_value)
+			 ELSE MAX (datetime_bin::date)
+			 END 
+			 as last_reported,
+			CASE
+			WHEN  MAX (datetime_bin::date)>= (insert_value) then 'True'::text
```

Comment:
this should be kept as boolean because at the very end it got casted back to boolean